src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs

```diff
@@ -42,53 +42,62 @@ internal void Load(IDictionary envVariables)
         {
             var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
-            foreach (DictionaryEntry entry in envVariables)
+            IDictionaryEnumerator e = envVariables.GetEnumerator();
+            try
```

Comment:
why are you adding try/finally?

Reply:
`foreach` generates one.  Since I stopped using foreach, I added the try/finally manually as the foreach would have.